### PR TITLE
Error with exception on SSL error

### PR DESCRIPTION
**Pull request by MorrisJobke**
Found while debugging a SSL error that happened a few lines below that.

### CODE REVIEW

**Review by rullzer — APPROVED** [review]

**Review by GretaD — APPROVED** [review]

File: lib/private/Authentication/Token/PublicKeyTokenProvider.php [hunk 1]
```diff
@@ -306,6 +306,7 @@ private function newToken(string $token,
 		$res = openssl_pkey_new($config);
 		if ($res === false) {
 			$this->logOpensslError();
+			throw new \Exception('OpenSSL error happened');
```
Comment (kesselb) [inline lib/private/Authentication/Token/PublicKeyTokenProvider.php]: Mind to pick a more specific exception?

Comment (MorrisJobke) [reply lib/private/Authentication/Token/PublicKeyTokenProvider.php]: Anything in mind?

Comment (kesselb) [inline lib/private/Authentication/Token/PublicKeyTokenProvider.php]:
```suggestion
			throw new \RuntimeException('OpenSSL reported a problem');
```

Comment (kesselb) [reply lib/private/Authentication/Token/PublicKeyTokenProvider.php]: https://github.com/nextcloud/server/blob/f03eb7ec3c130d19323f7fb4bdb5ba398f1b3e2d/lib/private/Authentication/Token/IProvider.php#L37-L49

Please add the exception to the interface.

**Comment by MorrisJobke** [comment]
@kesselb Adjusted

File: lib/private/Authentication/Token/PublicKeyTokenProvider.php [hunk 2]
```diff
@@ -61,6 +61,9 @@ public function __construct(PublicKeyTokenMapper $mapper,
 		$this->time = $time;
 	}
 
+	/**
+	 * @throws \RuntimeException when OpenSSL reports a problem
```
Comment (kesselb) [inline lib/private/Authentication/Token/PublicKeyTokenProvider.php]: image

Could you remove it here again? PHPStorm is not happy ;)

File: lib/private/Authentication/Token/IProvider.php [hunk 3]
```diff
@@ -158,6 +160,7 @@ public function setPassword(IToken $token, string $tokenId, string $password);
 	 * @param string $oldTokenId
 	 * @param string $newTokenId
 	 * @return IToken
+	 * @throws \RuntimeException when OpenSSL reports a problem
```
Comment (kesselb) [inline lib/private/Authentication/Token/IProvider.php]:
```suggestion
	 * @throws InvalidTokenException
```

File: lib/private/Authentication/Token/PublicKeyTokenProvider.php (hunk 2 above)
Comment (MorrisJobke) [reply lib/private/Authentication/Token/PublicKeyTokenProvider.php]: Obviously there was more missing, but I think that this is better than not document it.

File: lib/private/Authentication/Token/IProvider.php (hunk 3 above)
Comment (MorrisJobke) [reply lib/private/Authentication/Token/IProvider.php]: Why? Sure the InvalidTokenException is missing here as well, but also the RuntimeException. I just documented the exception on all public API endpoints that may run into it.

File: lib/private/Authentication/Token/PublicKeyTokenProvider.php (hunk 2 above)
Comment (kesselb) [reply lib/private/Authentication/Token/PublicKeyTokenProvider.php]:
```
    /**
     * {@inheritdoc}
     */
```

would be more correct then.

File: lib/private/Authentication/Token/IProvider.php (hunk 3 above)
Comment (kesselb) [reply lib/private/Authentication/Token/IProvider.php]: Hmm. Where is `\RuntimeException` thrown from this code path?

Comment (kesselb) [reply lib/private/Authentication/Token/IProvider.php]: Found it ;)

File: lib/private/Authentication/Token/PublicKeyTokenProvider.php (hunk 2 above)
Comment (MorrisJobke) [reply lib/private/Authentication/Token/PublicKeyTokenProvider.php]: Yep - fixed.

**Review by kesselb — APPROVED** [review]

**Pull request merged**